packages/core/src/main/mls/MLSService/MLSService.ts

```diff
@@ -59,41 +59,34 @@ export class MLSService {
     return client;
   }
 
-  private async uploadCommitBundle(groupIdDecodedFromBase64: Uint8Array, commitBundle: CommitBundle) {
+  private async uploadCommitBundle(groupId: Uint8Array, commitBundle: CommitBundle) {
     const coreCryptoClient = this.getCoreCryptoClient();
 
     if (commitBundle.welcome) {
       //@todo: it's temporary - we wait for core-crypto fix to return the actual Uint8Array instead of regular array
       await this.apiClient.api.conversation.postMlsWelcomeMessage(optionalToUint8Array(commitBundle.welcome));
     }
     if (commitBundle.commit) {
-      const messageResponse = await this.apiClient.api.conversation.postMlsMessage(
-        //@todo: it's temporary - we wait for core-crypto fix to return the actual Uint8Array instead of regular array
-        optionalToUint8Array(commitBundle.commit),
-      );
-      await coreCryptoClient.commitAccepted(groupIdDecodedFromBase64);
-      return messageResponse;
+      try {
+        const messageResponse = await this.apiClient.api.conversation.postMlsMessage(
+          //@todo: it's temporary - we wait for core-crypto fix to return the actual Uint8Array instead of regular array
+          optionalToUint8Array(commitBundle.commit),
+        );
+        await coreCryptoClient.commitAccepted(groupId);
+        return messageResponse;
+      } catch (error) {
+        await coreCryptoClient.clear_pending_commit(groupId);
+        return null;
```

Comment:
These are new bit. This is how we should handle backend errors when committing something